### DISCUSSION TITLE
Is there any example of code to init provider?

### DISCUSSION THREAD

**Discussion by alcyoneus-g** [reach-sh/reach-lang#1082] [Q&A] [answered]
https://docs.reach.sh/frontend/#p_36

```
interface Provider {
  algodClient: algosdk.Algodv2,
  indexer: algosdk.Indexer,
  getDefaultAddress: () => Promise<Address>,
  isIsolatedNetwork: boolean,
  signAndPostTxns: (txns:WalletTransaction[], opts?: any) => Promise<any>,
};
```

What is `signAndPostTxns` for and is there any example code to set `signAndPostTxns` and `getDefaultAddress`?

**Comment by jeapostrophe** [comment, marked as answer]
The text beneath that explains what `signAndPostTxns` is for. You can find examples by looking in https://github.com/reach-sh/reach-lang/blob/master/js/stdlib/ts/ALGO.ts for `makeProviderBy`. This is an internal function and we do not intend to put an example in the docs.
